### PR TITLE
add exported wrapper for hashToField

### PR DESCRIPTION
**Pull request by tatianab**
@armfazh

### CODE REVIEW

File: hash.go [hunk 1]
```diff
@@ -17,61 +17,86 @@ type HashToPoint interface {
...
 func (s *encodeToCurve) IsRandomOracle() bool { return false }
 func (s *encodeToCurve) Hash(in []byte) C.Point {
-	u := s.hashToField(in, 1)
+	u := s.FieldEncoding.hashToField(in, 1)
 	Q := s.Mapping.Map(u[0])
 	P := s.E.ClearCofactor(Q)
 	return P
 }
 
+func (s *encodeToCurve) GetHashToScalar() HashToScalar {
```
Comment (armfazh) [inline hash.go]: The receiver of this method must be `encoding` as is shared between `encodeToCurve` and `hashToCurve` .

File: hash.go [hunk 2]
```diff
@@ -17,61 +17,86 @@ type HashToPoint interface {
...
+	return s.FieldEncoding
```
Comment (armfazh) [inline hash.go]: However, the object returned must not be the field of the curve, but the field of the scalars.
You need to define the fields using the order of the curve as the prime modulus.

File: hash.go [hunk 3]
```diff
@@ -17,61 +17,86 @@ type HashToPoint interface {
...
 			vj := new(big.Int).SetBytes(t)
 			v[j] = vj.Mod(vj, p)
 		}
-		u[i] = F.Elt(v)
+		u[i] = f.F.Elt(v)
 	}
 	return u
 }
 
+type encoding struct {
```
Comment (armfazh) [inline hash.go:66]: You can augment this structure with a member that is the field of scalars.

**Review by armfazh — CHANGES_REQUESTED** [review]
The general comment is that the field of the scalars is a different object from the field where the curve is defined.
You can find the prime modulus `h` in each curve definition.

File: suites.go [hunk 4]
```diff
@@ -48,12 +49,14 @@ func (id SuiteID) Get(dst []byte) (HashToPoint, error) {
...
 			E: E,
-			FieldEncoding: &fieldEncoding{
+			Field: &fieldEncoding{
 				F:   E.Field(),
 				Exp: exp,
 				L:   s.L,
 			},
-			Mapping: m,
+			// TATIANA: what should name be here?
+			ScalarField: field.NewFp("", E.Field().P()),
```
Comment (armfazh) [inline suites.go]: This is not the modulus of the scalar field: `E.Field().P()`, it should be the order of the curve.

File: go.sum [hunk 5]
```diff
@@ -1,5 +1,7 @@
 github.com/armfazh/tozan-ecc v0.1.1 h1:t22NORZiIzJOUdNN/bKdRJpfZOSLohoCSUkivm6+hIE=
 github.com/armfazh/tozan-ecc v0.1.1/go.mod h1:u25eZC5Z8uJFQxJxGBz1Blfii/7m3DfmwX0vFnwtG9I=
+github.com/armfazh/tozan-ecc v0.1.2 h1:mhwrQnxiKqlDUdU5v6o6IBXF+CUPhuLAEIAH1gdIAf0=
```
Comment (armfazh) [inline go.sum:1]: run go  mod tidy

Comment (tatianab) [reply go.sum:1]: done

**Review by armfazh — APPROVED** [review]

**Pull request merged**